src/main/java/producer/SerializedProducerRecord.java

```diff
@@ -0,0 +1,45 @@
+package producer;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import commons.header.Header;
+import commons.headers.Headers;
+import commons.serializers.OptionalSerializer;
+import commons.serializers.HeaderSerializer;
+import commons.serializers.HeadersSerializer;
+import commons.serializers.ProducerRecordSerializer;
+
+import java.util.Optional;
+
+public class SerializedProducerRecord {
+    private static Kryo kryo;
+
+    static {
+        // All classes will share a single Kryo object.
+        kryo = new Kryo();
+
+        // Kyro requires each class to be registered for serialization. (Includes nested objects)
+        kryo.register(ProducerRecord.class, new ProducerRecordSerializer<>(String.class, Integer.class));
+        kryo.register(Optional.class, new OptionalSerializer());
+        kryo.register(Headers.class, new HeadersSerializer());
+        kryo.register(Header.class, new HeaderSerializer());
+    }
+
+    public static <K,V> byte[] serialize(ProducerRecord<K,V> record) {
+        Output output = new Output(4096, -1); // 4 KB is the typical size of memory page
```

Comment:
could be nice to have this as a constant

Reply:
@KyoshiNoda constant as in `4096` is in it's own variable, something like `INITIAL_BUFFER_SIZE = 4096`

Reply:
> @KyoshiNoda constant as in `4096` is in it's own variable, something like `INITIAL_BUFFER_SIZE = 4096`

oops lol I'll fix it right now 😭